core/src/main/java/nl/nn/adapterframework/validation/XercesXmlValidator.java

```diff
@@ -282,18 +298,28 @@ public ValidatorHandler getValidatorHandler(IPipeLineSession session, Validation
 		}
 		return validatorHandler;
 	}
-	public XMLReader createValidatingParser(IPipeLineSession session, ValidationContext context) throws XmlValidatorException, ConfigurationException, PipeRunException {
-		SymbolTable symbolTable = ((XercesValidationContext)context).getSymbolTable();
-		XMLGrammarPool grammarPool = ((XercesValidationContext)context).getGrammarPool();
 
-		XMLReader parser = new SAXParser(new ShadowedSymbolTable(symbolTable), grammarPool);
+	@Override
+	public String validate(Object input, IPipeLineSession session, String logPrefix, Set<List<String>> rootValidations, Map<List<String>, List<String>> invalidRootNamespaces, boolean resolveExternalEntities) throws XmlValidatorException, PipeRunException, ConfigurationException {
+		ValidationContext context = createValidationContext(session, rootValidations, invalidRootNamespaces);
+		try {
+			ValidatorHandlerImpl validatorHandler = getValidatorHandler(session, context);
+			SAXSource inputSource = XmlUtils.stringToSAXSource(input.toString(), true, false);
+			validatorHandler.validate(inputSource, null);
+		} catch (SAXException | IOException e) {
+			return finalizeValidation(context, session, e);
+		}
+		return finalizeValidation(context, session, null);
+	}
```

Comment:
Why is it necessary to override validate() here?
If not necessary, please remove it.
If you want to keep it, please avoid assuming 'input' is a string.

Reply:
Because AbstractValidator's validate uses `createValidationContext `to get a XMLReader and uses `reader.parse` method to validate it. However, with new Xerces we are using validator handler, and it does not extend/implement XmlReader, so I could not return xml reader to use AbstractValidator's validate method, I needed to override it.
I will modify it to not assume input is a string.

Reply:
OK, but this means this is only a partial implementation of XSD 1.1 support. The method createValidatingParser should also return a parser that validates agains XSD 1.1.
This is used for instance in the Json2XmlValidator, for conversions from XML to JSON.
To extend your test cases to this, I suggest you migrate the tests you have built in _XmlValidatorExtraTest_ to _XmlValidatorTestBase_. Then it will be used also in the Json2XmlValidatorTest, that uses the validating parser in Json2XmlValidator.

Reply:
Restructured the validator. Now, we support full implementation of xsd 1.1